src/vm/core.zig

```diff
@@ -305,11 +305,12 @@ pub const CairoVM = struct {
         instruction: *const instructions.Instruction,
     ) !OperandsResult {
         var op_res = OperandsResult.default();
-
         op_res.res = null;
 
         op_res.dst_addr = try self.run_context.computeDstAddr(instruction);
-        op_res.dst = try self.segments.memory.get(op_res.dst_addr);
+
+        op_res.dst = self.segments.memory.get(op_res.dst_addr) catch null;
```

Comment:
Here I'm not sure that we should allow our final result `op_res.dst` to be `null`. I think that's why in the Rust implemented they inserted an intermediary variable here:

https://github.com/lambdaclass/cairo-vm/blob/e6171d66a64146acc16d5512766ae91ae044f297/vm/src/vm/vm_core.rs#L557

And the final result is `MaybeRelocatable` and not `Option<MaybeRelocatable>`: https://github.com/lambdaclass/cairo-vm/blob/e6171d66a64146acc16d5512766ae91ae044f297/vm/src/vm/vm_core.rs#L37-L42

I modified the function this way here: https://github.com/keep-starknet-strange/cairo-zig/pull/167

Reply:
yeah, i made these changes just to sanity check my impl. i removed them to give way to your fixes np!